src/System.Data.SqlClient/tests/Tools/TDS/TDS.EndPoint/SSPI/SecurityWrapper.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.SqlServer.TDS.EndPoint.SSPI
+{
+    /// <summary>
+    /// Wrapper for security functions that implement SSPI
+    /// </summary>
+    internal class SecurityWrapper
```

Comment:
SSPI/SecurityWrapper.cs is being used to validate the authentication token being sent to the server and it will be used in test cases where we set IntegratedSecurity=true in Sql Connection string. 
We are still working on the support for IntegratedSecurity=true on Unix. 

We were considering using System.Net.Security for bringing in Integrated Auth for SqlClient on Unix, but the current API surface don't provide for our requirements. 

System.Net.Security provides NegotiateStream which then calls Interop for SSPI. The closest abstraction that we could find, was in NegotiateStreamPal which is not exposed in the public surface area. 

For the TDS protocol, the client gets the auth token from the Native layer and put it in the TDS packet. The server then uses Native calls to validate this token. That's the reason we are working with SSPI directly. 

For Unix, we throw a PlatformNotSupported in the SqlClient and also in the TDS Server. We will add this once we figure out the Unix Integrated Security approach.